src/xmipp/libraries/reconstruction/shift_particles.cpp

```diff
@@ -0,0 +1,128 @@
+// /***************************************************************************
+//  *
+//  * Authors:    Estrella Fernandez Gimenez (me.fernandez@cnb.csic.es)
+//  *
+//  * Unidad de  Bioinformatica of Centro Nacional de Biotecnologia , CSIC
+//  *
+//  * This program is free software; you can redistribute it and/or modify
+//  * it under the terms of the GNU General Public License as published by
+//  * the Free Software Foundation; either version 2 of the License, or
+//  * (at your option) any later version.
+//  *
+//  * This program is distributed in the hope that it will be useful,
+//  * but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  * GNU General Public License for more details.
+//  *
+//  * You should have received a copy of the GNU General Public License
+//  * along with this program; if not, write to the Free Software
+//  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+//  * 02111-1307  USA
+//  *
+//  *  All comments concerning this program package may be sent to the
+//  *  e-mail address 'xmipp@cnb.csic.es'
+//  ***************************************************************************/
+
+ #include "shift_particles.h"
+ #include "core/transformations.h"
+ #include "core/xmipp_image_extension.h"
+ #include "core/xmipp_image_generic.h"
+ #include "core/xmipp_fft.h"
+ #include "core/xmipp_fftw.h"
+ #include "data/projection.h"
+ #include "data/mask.h"
+ #include <iostream>
+ #include <string>
+ #include <sstream>
+ #include "data/image_operate.h"
+
+
+ // Read arguments ==========================================================
+ void ProgShiftParticles::readParams()
+ {
+	fnParticles = getParam("-i");
+	x0=getDoubleParam("--center",0);
+	y0=getDoubleParam("--center",1);
+	z0=getDoubleParam("--center",2);
+	boxSize=getIntParam("--boxSize");
+	fnOut=getParam("-o");
+	if (fnOut=="")
+		fnOut="output_particle_";
+ }
+
+ // Show ====================================================================
+ void ProgShiftParticles::show()
+ {
+    if (!verbose)
+        return;
+	std::cout
+	<< "Input particles:   	" << fnParticles << std::endl
+	<< "New center:  		" << x0 << ", "  << y0 << ", "  << z0 << std::endl
+	<< "Output box size: 	" << boxSize     << std::endl
+	<< "Output particles: 	" << fnOut 	     << std::endl
+	;
+ }
+
+ // usage ===================================================================
+ void ProgShiftParticles::defineParams()
+ {
+     //Usage
+     addUsageLine("Center particles into a selected point of a volume.");
+     //Parameters
+     addParamsLine("-i <particles>         			: Particles metadata (.xmd file)");
+     addParamsLine("[-o <structure=\"\">]  			: Output filename suffix for shifted particles");
```

Comment:
so do you expect that `-o` includes the extension or not?
`-o` is typically used for full path name